File: dashboard_etapa/tools/list_projects.js
const https = require('https');
const fs = require('fs');
const path = require('path');

// Cargar .env manualmente
function loadEnv() {
    try {
        const envPath = path.join(__dirname, '..', '.env');
        const content = fs.readFileSync(envPath, 'utf8');
        const env = {};
        content.split('\n').forEach(line => {
            if (line.includes('=') && !line.startsWith('#')) {
                const [key, value] = line.trim().split('=', 2);
                env[key] = value;
            }
        });
        return env;
    } catch (e) {
        console.error("Error loading .env:", e.message);
        process.exit(1);
    }
}

const config = loadEnv();
const token = config.SUPABASE_ACCESS_TOKEN;

if (!token) {
    console.error("Error: SUPABASE_ACCESS_TOKEN not found in .env");
    process.exit(1);
}

const options = {
    hostname: 'api.supabase.com',
    path: '/v1/projects',
    method: 'GET',
    headers: {
        'Authorization': `Bearer ${token}`,
        'User-Agent': 'IAmazing-Dashboard/1.0'
    }
};

console.log("Connecting to Supabase Management API...");

const req = https.request(options, (res) => {
    let data = '';

    res.on('data', (chunk) => {
        data += chunk;
    });

    res.on('end', () => {
        if (res.statusCode === 200) {
            try {
                const projects = JSON.parse(data);
                console.log(`SUCCESS: Found ${projects.length} projects.`);
                console.log("-".repeat(40));
                projects.forEach(p => {
                    console.log(`ID: ${p.id}`);
                    console.log(`Name: ${p.name}`);
                    console.log(`Status: ${p.status}`);
                    console.log(`Region: ${p.region}`);
                    console.log("-".repeat(40));
                });
            } catch (e) {
                console.error("Error parsing JSON:", e.message);
            }
        } else {
            console.error(`Error: HTTP ${res.statusCode}`);
            console.error(data);
        }
    });
});

req.on('error', (e) => {
    console.error(`Request error: ${e.message}`);
});

req.end();

File: dashboard_etapa/tools/list_projects.py
import os
import json
import urllib.request

# Cargar token desde .env simple (sin librerías externas para máxima compatibilidad)
def load_env():
    env = {}
    try:
        with open('../.env', 'r') as f:
            for line in f:
                if '=' in line and not line.startswith('#'):
                    key, value = line.strip().split('=', 1)
                    env[key] = value
    except FileNotFoundError:
        print("Error: .env file not found")
        exit(1)
    return env

config = load_env()
token = config.get('SUPABASE_ACCESS_TOKEN')

if not token:
    print("Error: SUPABASE_ACCESS_TOKEN not found in .env")
    exit(1)

url = "https://api.supabase.com/v1/projects"
req = urllib.request.Request(url)
req.add_header('Authorization', f'Bearer {token}')

print(f"Connecting to Supabase Management API...")

try:
    with urllib.request.urlopen(req) as response:
        if response.status == 200:
            data = json.loads(response.read().decode())
            print(f"SUCCESS: Found {len(data)} projects.")
            print("-" * 40)
            for project in data:
                print(f"ID: {project.get('id')}")
                print(f"Name: {project.get('name')}")
                print(f"Status: {project.get('status')}")
                print(f"Region: {project.get('region')}")
                print(f"Created: {project.get('created_at')}")
                print("-" * 40)
        else:
            print(f"Error: HTTP {response.status}")
except urllib.error.HTTPError as e:
    print(f"HTTP Error: {e.code} - {e.reason}")
    print(e.read().decode())
except Exception as e:
    print(f"Error: {e}")
